test/example/tutorials/test_decompositions.py

```diff
@@ -0,0 +1,221 @@
+import math
+
+import sympy as sp
+from IPython.display import display
+
+from opensquirrel.circuit_builder import CircuitBuilder
+from opensquirrel.decomposer.aba_decomposer import XZXDecomposer, ZYZDecomposer
+from opensquirrel.decomposer.cnot_decomposer import CNOTDecomposer
+from opensquirrel.decomposer.mckay_decomposer import McKayDecomposer
+from opensquirrel.default_gates import H, Rx, Rz
+from opensquirrel.ir import Float, Qubit
+
+
+class TestDecomposition:
+    def test_sympy_disp(self):
+
+        theta1, theta2, theta3 = sp.symbols("theta_1 theta_2 theta_3")
+
+        z1 = sp.algebras.Quaternion.from_axis_angle((0, 0, 1), theta1)
+        y = sp.algebras.Quaternion.from_axis_angle((0, 1, 0), theta2)
+        z2 = sp.algebras.Quaternion.from_axis_angle((0, 0, 1), theta3)
+
+        rhs = sp.trigsimp(sp.expand(z1 * y * z2))
+
+        p, m = sp.symbols("p m")
+
+        rhs_simplified = rhs.subs({theta1 + theta3: p, theta1 - theta3: m})
+
+        alpha, nx, ny, nz = sp.symbols("alpha n_x n_y n_z")
+
+        q = sp.algebras.Quaternion.from_axis_angle((nx, ny, nz), alpha).subs(
+            {nx**2 + ny**2 + nz**2: 1}  # We assume the axis is normalized.
+        )
+
+        display(
+            sp.Eq(rhs_simplified.a, q.a),
+            sp.Eq(rhs_simplified.b, q.b),
+            sp.Eq(rhs_simplified.c, q.c),
+            sp.Eq(rhs_simplified.d, q.d),
+        )
+
+        sp.trigsimp(sp.Eq(rhs_simplified.a, q.a).subs(sp.cos(theta2 / 2), nz * sp.sin(alpha / 2) / sp.sin(p / 2)))
+
+        theta1, theta2, theta3 = sp.symbols("theta_1 theta_2 theta_3")
+
+        y1 = sp.algebras.Quaternion.from_axis_angle((0, 1, 0), theta1)
+        z = sp.algebras.Quaternion.from_axis_angle((0, 0, 1), theta2)
+        y2 = sp.algebras.Quaternion.from_axis_angle((0, 1, 0), theta3)
+
+        rhs = sp.trigsimp(sp.expand(y1 * z * y2))
+        rhs_string = "cos(theta_2/2)*cos((theta_1 + theta_3)/2) + sin(theta_2/2)*sin((theta_1 - theta_3)/2)*i + sin((theta_1 + theta_3)/2)*cos(theta_2/2)*j + sin(theta_2/2)*cos((theta_1 - theta_3)/2)*k"
+        assert str(rhs) == rhs_string
+
+    def test_builder_ABA(self):
```

Comment:
What does `builder_ABA` means here? What is being tested? The builder? An input circuit for the ABA decomposer?

I would suggest to not use uppercase for function names.

Reply:
According to the names of the other tests, this should be `test_aba_decomposer`:

```suggestion
    def test_aba_decomposer(self):
```